model/src/main/kotlin/VulnerabilityReference.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import java.net.URI
+
+/**
+ * A data class representing detailed information about a vulnerability obtained from a specific source.
+ *
+ * A single vulnerability can be listed by multiple sources using different scoring systems to denote its severity.
+ * So when ORT queries different providers for vulnerability information it may well find multiple records for a single
+ * vulnerability, which could even contain contradicting information. To model this, a [Vulnerability] is associated
+ * with a list of references; each reference points to the source of the information and has some detail information
+ * provided by this source.
+ */
+data class VulnerabilityReference(
+    /**
+     * The URI pointing to details of this vulnerability. This can also be used to derive the source of this
+     * information.
+     */
+    val url: URI,
+
+    /**
+     * The name of the scoring system to express the severity of this vulnerability if available.
+     */
+    val scoringSystem: String?,
+
+    /**
+     * The severity assigned to the vulnerability by this reference. Note that this is a plain string, whose meaning
+     * depends on the concrete scoring system. It could be a number, but also a constant like _LOW_ or _HIGH_.
+     */
+    val severity: String?
```

Comment:
Why is this nullable? What does it mean if it's `null`?

Reply:
This is a good question. The [example responses](https://github.com/nexB/vulnerablecode/pull/369#issuecomment-796877606) we got from VulnerableCode contain some entries without a scoring system and a severity. Do you think we should filter those out or rather keep them with the meaning that a vulnerability exists, but no further information is available?

Reply:
Hmm, we probably shouldn't deliberately omit vulnerabilities that we know about. I was not aware that vulnerabilities without a unknown severity are a thing. Then please just extend the docs to explain that `null` means "unknown". And later we need to take care to map this to a proper string.